### PR TITLE
add a function and fix some errors

### PR DESCRIPTION
**Pull request by alexzshl**
1. add function: rename the name on task show on statusbar by edit task.json
2. fix error: when the arg's type is object,task hide don't work

### CODE REVIEW

File: js/extension.js [hunk 1]
```diff
@@ -86,6 +86,8 @@ function computeId(task, config) {
         for (var arg of args) {
             if (typeof arg == "string") {
                 props.push(arg);
+            } else if (typeof arg == "object") {
```
Comment (alexzshl) [inline js/extension.js]: when arg is object,this function get a wrong taskID witch diffirent from the taskID below

File: js/extension.js [hunk 2]
```diff
@@ -108,15 +111,16 @@ function loadTasks(context) {
             if (getValue2(task, config, "options", "statusbar") == "hide") {
                 hide[computeId(task, config)] = true;
             }
+            statusbarLabels[computeId(task, config)] = getValue2(task, config, "options", "statusbarLabel");
```
Comment (alexzshl) [inline js/extension.js]: add an item "statusbarLabel" to "options" of tasks.json,if it's configed,this extentsion use it as the task name show on statusbar

File: js/extension.js [hunk 3]
```diff
@@ -108,15 +111,16 @@ function loadTasks(context) {
...
         }
     }
 
     let version = vscode.version.split(".");
-    let priority = version[1] == 36? 50: 51;
+    let priority = version[1] == 37? 50: 51;
 
     vscode.tasks.fetchTasks().then((tasks)=>{
         for (const task of tasks) {
-            let name = task.name;
```
Comment (alexzshl) [inline js/extension.js]: delete this line

File: js/extension.js [hunk 4]
```diff
@@ -108,15 +111,16 @@ function loadTasks(context) {
...
+            let name = statusbarLabels[task.name + ',' + task.definition.id] || task.name;
```
Comment (alexzshl) [inline js/extension.js]: use statusbarLabel as task name at first

File: js/extension.js [hunk 5]
```diff
@@ -108,15 +111,16 @@ function loadTasks(context) {
             if (getValue2(task, config, "options", "statusbar") == "hide") {
                 hide[computeId(task, config)] = true;
             }
+            statusbarLabels[computeId(task, config)] = getValue2(task, config, "options", "statusbarLabel");
         }
     }
 
     let version = vscode.version.split(".");
-    let priority = version[1] == 36? 50: 51;
+    let priority = version[1] == 37? 50: 51;
```
Comment (actboy168) [inline js/extension.js]: This is to fix the 1.36-specific bug, please don't change it.

**Comment by actboy168** [comment]
It looks good to me. But split the bug fix and add features into two pull requests.

**Comment by alexzshl** [comment]
好的.但是我升级到1.37之后,所有的任务向左偏移了,超过了错误图标和警告图标,是否可以修复呢?


-------------------------------------------------------------
From：alex
email：zd870905@126.com

On 8/20/2019 10:17，actboy168<notifications@github.com> wrote：

@actboy168 requested changes on this pull request.

In js/extension.js:

>          }
     }
 
     let version = vscode.version.split(".");
-    let priority = version[1] == 36? 50: 51;
+    let priority = version[1] == 37? 50: 51;


This is to fix the 1.36-specific bug, please don't change it.

**Comment by actboy168** [comment]
这个和1.35以及之前的表现是一致的

File: js/extension.js (hunk 5 above)
Comment (alexzshl) [reply js/extension.js]: 为了避免后续版本更迭导致task位置异常,我建议修改为如下代码
`let priority = version[1] >= 36? 50: 51;`
不知是否可行?

File: js/extension.js [hunk 6]
```diff
@@ -108,15 +111,16 @@ function loadTasks(context) {
             if (getValue2(task, config, "options", "statusbar") == "hide") {
                 hide[computeId(task, config)] = true;
             }
+            statusbarLabels[computeId(task, config)] = getValue2(task, config, "options", "statusbarLabel");
         }
     }
 
     let version = vscode.version.split(".");
-    let priority = version[1] == 36? 50: 51;
+    let priority = version[1] >= 36? 50: 51;
```
Comment (alexzshl) [inline js/extension.js]: fixed

File: package.json [hunk 7]
```diff
@@ -25,5 +25,13 @@
     "activationEvents": [
         "workspaceContains:.vscode/tasks.json"
     ],
-    "main": "./js/extension"
+    "main": "./js/extension",
+    "contributes": {
+        "jsonValidation": [
```
Comment (alexzshl) [inline package.json]: add jshint

**Comment by actboy168** [comment]
statusbar的优先级，不能称之为异常，这是一直以来的行为。我会考虑更好的方案，但不能在这个PR里修改。

**Pull request merged**